### PR TITLE
Allow `inputs_for` to skip hidden fields for more control over the final markup

### PR DESCRIPTION
**Pull request by LostKobrakai**
For #258

### CODE REVIEW

File: lib/phoenix_html/form.ex [hunk 1]
```diff
@@ -423,11 +430,20 @@ defmodule Phoenix.HTML.Form do
...
+        else
+          [hidden_fields_of_form(form), fun.(form)]
+        end
       end)
     )
   end
+  
+  @doc false
+  @spec hidden_fields_of_form(t) :: [Phoenix.HTML.safe()]
+  def hidden_fields_of_form(%{hidden: hidden}) do
```
Comment (josevalim) [inline lib/phoenix_html/form.ex]: Shall we call it `hidden_form_inputs/1`?

**Comment by LostKobrakai** [comment]
I've renamed the helper. For testing the new functionality we would need a new implementation of `Phoenix.HTML.FormData` though. `Plug.Conn` doesn't use any hidden fields.

**Comment by josevalim** [comment]
Let’s do that then! We can pass the hidden fields as an option to this new
implementation.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

File: lib/phoenix_html/form.ex [hunk 2]
```diff
@@ -410,10 +410,17 @@ defmodule Phoenix.HTML.Form do
       applies if the field value is a list and no parameters were
       sent through the form.
 
+    * `:skip_hidden` - skip the automatic rendering of hidden
```
Comment (feliperenan) [inline lib/phoenix_html/form.ex]: We could call this option of `skip_hidden_inputs` to be more explicit. Wdyt?

**Comment by feliperenan** [comment]
Could we only add the `hidden` option in the `Phoenix.HTML.FormData`? I mean something like that:

```Elixir
def to_form(conn_or_atom, form, field, opts) when is_atom(field) or is_binary(field) do
  #...
  {hidden, opts} = Keyword.pop(opts, :hidden, [])
  #...

  cond do
    # cardinality: one
    is_map(default) ->
      [
        %Phoenix.HTML.Form{
          source: conn_or_atom,
          impl: __MODULE__,
          id: id,
          name: name,
          data: default,
          params: params || %{},
          hidden: hidden,
          options: opts
        }
      ]

    #...
  end
  #...
end
```

https://github.com/phoenixframework/phoenix_html/blob/837fdbf2712bb2f38dcd3dc0e828847a2d49a39b/lib/phoenix_html/form_data.ex#L88

I know that we would be doing that only because the test, but would be easier than creating a new implemention (I guess).

If so, we could add these tests :).

```Elixir
describe "inputs_for/4" do
  test "generate a new form builder for the given parameter" do
    conn = conn()

    form =
        form_for(conn, "/", [as: :user], fn form ->
          inputs_for(form, :company, fn company_form ->
            text_input company_form, :name
          end)
        end)
        |> safe_to_string()

    assert form =~ ~s(<input name="_utf8" type="hidden" value="‚úì">)
    assert form =~ ~s(<input id="user_company_name" name="user[company][name]" type="text">)
  end

  test "generate a new form builder with hidden id fields when there are present" do
    conn = conn()

    form =
        form_for(conn, "/", [as: :user], fn form ->
          inputs_for(form, :company, [hidden: [id: 1]],fn company_form ->
            text_input company_form, :name
          end)
        end)
        |> safe_to_string()

    assert form =~ ~s(input id="user_company_id" name="user[company][id]" type="hidden" value="1">)
    assert form =~ ~s(<input id="user_company_name" name="user[company][name]" type="text">)
  end

  test "skip hidden inputs" do
    conn = conn()

    form =
        form_for(conn, "/", [as: :user], fn form ->
          inputs_for(form, :company, [skip_hidden: true, hidden: [id: 1]],fn company_form ->
            text_input company_form, :name
          end)
        end)
        |> safe_to_string()

    refute form =~ ~s(input id="user_company_id" name="user[company][id]" type="hidden" value="1">)
    assert form =~ ~s(<input id="user_company_name" name="user[company][name]" type="text">)
  end
end
```

@LostKobrakai are you willing to continue working on this? Otherwise I can open another pull-request to finish this work :).

**Comment by LostKobrakai** [comment]
@feliperenan go ahead if you like.

**Comment by feliperenan** [comment]
I've opened another one https://github.com/phoenixframework/phoenix_html/pull/259 using your commit as reference, we could close this one now. 😅

**Pull request closed without merging**